File: 1560. Most Visited Sector in a Circular Track/Solution.cpp
#include <vector>
using namespace std;

class Solution
{
public:
    vector<int> mostVisited(int n, vector<int> &rounds)
    {
        int bg = rounds[0], ed = rounds.back();
        vector<int> res;
        if (bg <= ed) {
            for (int i = bg; i <= ed; i++)
                res.emplace_back(i);
        } else {
            for (int i = 1; i <= ed; i++)
                res.emplace_back(i);
            for (int i = bg; i <= n; i++)
                res.emplace_back(i);
        }
        return res;
    }
};

File: 1560. Most Visited Sector in a Circular Track/Solution.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<Integer> mostVisited(int n, int[] rounds) {
        List<Integer> res = new ArrayList<>();
        int bg = rounds[0], ed = rounds[rounds.length - 1];
        if (bg <= ed) {
            for (int i = bg; i <= ed; i++) {
                res.add(i);
            }
        } else {
            for (int i = 1; i <= ed; i++) {
                res.add(i);
            }
            for (int i = bg; i <= n; i++) {
                res.add(i);
            }
        }
        return res;
    }
}
